Automattic/jetpack
# Add/segmentation to jetpack products my jetpack

## Proposed changes:

1. Split products up on the My Jetpack page between owned and unowned
Note that this will be different from products that have an "active" versus "inactive" status. A user may be considered as "having" or "owning" a product if any of the following are true:

* They have a standalone plugin for that product (the product may still need a plan)
* They have a plan for a product (even if the plugin is missing)
* They had a product that was working before but is not working anymore now due to a connection error (part of the work that is happening with the user connection delay project: https://github.com/Automattic/jetpack-roadmap/issues/1210)

2. Sort products in the "owned section" by active -> warning -> inactive since all 3 statuses can exist in an "owned" state

### Other information:


## Jetpack product discussion

P2: p1HpG7-sMW-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Create a JN site with Jetpack Beta installed but no branches selected or plugins installed
2. Try installing a standalone plugin to start and go to My Jetpack. Make sure to check out this branch in Jetpack beta for that plugin
3. You will likely see Akismet in your owned products since that mostly goes off API key and whether your user account has a plan. Make sure they match the design comps (g1ZLnS65PzfOL9cmByykvf-fi-4412_2) and that you only see the plugin you installed and Akismet in the "My Products" section (or possibly only VideoPress)
![image](https://github.com/Automattic/jetpack/assets/65001528/d01504d5-debf-4efa-82e8-912bd0dc4267)
4. Now install the main Jetpack plugin (again with this branch checked out)
5. Nothing should change in My Jetpack just yet
![image](https://github.com/Automattic/jetpack/assets/65001528/540fb4ba-ca69-4f4b-ab61-96984e4b0f5c)
6. Now connect your site to Jetpack (no user connection)
7. You should see Stats and Creator now in the Owned products section
![image](https://github.com/Automattic/jetpack/assets/65001528/c181533d-0c87-4c02-97d3-3b9a8f511701)
8. Now connect your user account, it should look roughly the same
![image](https://github.com/Automattic/jetpack/assets/65001528/33c0407d-b41b-434a-98e8-8d2e8e615605)
9. Now add a plan to your site from SA that is not currently "owned" on your site. When you go back to My Jetpack, that product should be shown in the "owned" section now
![image](https://github.com/Automattic/jetpack/assets/65001528/76a5772e-58de-4f5b-873e-80c1319d1802)
10. Install but don't activate a Jetpack plugin that is not currently owned by your site
11. The owned section should now have the new plugin as "inactive" at the end of the list of plugins
![image](https://github.com/Automattic/jetpack/assets/65001528/f79c2b4f-926d-41f5-aa5f-26c4c4284a18)
12. Throughout this process, make sure that the products in the "active" section are in the order of active -> warning (connection or plugin missing) -> inactive
13. Now disconnect your site
14. The "owned" section should have the same products as before with one exception
 * If it was "owned" solely because you had a plan. With a disconnected site, My Jetpack won't be able to tell you have a plan and the plugin was not active so it's not tracked in "historically_active_modules"
![image](https://github.com/Automattic/jetpack/assets/65001528/a4e8740d-1cba-4f75-88a6-50f3252788f5)
15. Try and combination of plugins and active/inactive/connection-issue, whatever you can think of to make sure this section always looks correct.
